delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/client/BlazegraphClient.scala

```diff
@@ -36,6 +39,19 @@ class BlazegraphClient(
   private val defaultProperties =
     ClasspathResourceUtils.ioPropertiesOf("blazegraph/index.properties").hideErrors.memoizeOnSuccess
 
+  override def query[R <: SparqlQueryResponse](
+      indices: Iterable[String],
+      q: SparqlQuery,
+      responseType: Aux[R],
+      additionalHeaders: Seq[HttpHeader]
+  ): IO[SparqlClientError, R] = {
+    val headers = queryTimeout match {
+      case finite: FiniteDuration => additionalHeaders :+ RawHeader(timeoutHeader, finite.toMillis.toString)
+      case _                      => additionalHeaders
+    }
+    super.query(indices, q, responseType, headers)
```

Comment:
If we look here: https://github.com/BlueBrain/nexus/blob/master/delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/http/HttpClientWorthRetry.scala#L24
And from the config, it looks like the canceled query will be retried 5 times before giving up.
It can be part of a second PR but shouldn't we have a dedicated client for user queries where we never retry.
The indexing streams would also not share the connection pool with the user related queries

Reply:
I created a separate issue for that https://github.com/BlueBrain/nexus/issues/2530